barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -346,14 +344,17 @@ template <typename Curve> class ShpleminiVerifier_ {
                                                  scalars,
                                                  constant_term_accumulator);
 
-        // Add contributions from A₀(r) and A₀(-r) to constant_term_accumulator:
+        gemini_evaluations[0] += p_neg;
```

Comment:
this sequence of operations with `gemini_evaluations`, `full_a_0_pos`, `p_pos` and `p_neg` is confusing. If there's no way to simplify what's happening here can you add some comments explaining what's going on? E.g. why do we need to add then subtract `p_neg` from `gemini_evaluations[0] `before and after calling `compute_gemini_batched_univariate_evaluation`?

Reply:
added comments